core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Client.java

```diff
@@ -22,6 +22,8 @@
 
 public interface OAuth2Client
 {
+    void loadConfiguration();
```

Comment:
This method does not belong here. None of the callees of that component should not be able to call that method or even know it exists.

```

        newSetBinder(binder, LoadableComponent.class).addBinding().to(YourOauth2Loader.class.orInstance);
    }

    interface LoadableComponent
    {
        void load();
    }

    {
        Injector injector = null;
        injector.getInstance(Key.get(new TypeLiteral<Set<LoadableComponent>>() {})).forEach(LoadableComponent::load);
    }
```

Reply:
Shouldn't the implementation of this method be just a part of a constructor?

Reply:
@kokosing I agree that this would look better but the problem is that `OAuth2Client` uses optional binding:
```java
        newOptionalBinder(binder, OAuth2Client.class)
                .setDefault()
                .to(NimbusClient.class)
                .in(Scopes.SINGLETON);
```
and with `LoadableComponent` binding it wouldn't be optional anymore. Or more correctly, I don't know to express this relationship using guice.